bundle/edu.gemini.qpt.client/src/main/java/edu/gemini/qpt/core/listeners/LimitsListener.java

```diff
@@ -93,6 +139,10 @@ public void propertyChange(PropertyChangeEvent evt) {
 			final Double maxAirmass = a.getMax(Circumstance.AIRMASS, false);
 			final Double minAirmass = a.getMin(Circumstance.AIRMASS, false);
 
+			final ElevationConstraintSolver elevationSolver = ElevationConstraintSolver.forObs(site, a.getObs());
+
+            final Option<Interval> elevationInterval = night.map(n -> elevationSolver.solve(n).getIntervals().first());
```

Comment:
Is it guaranteed that the `elevationSolver.solve(n).getIntervals()` will be nonempty?  Otherwise `.first()` will throw an exception.